### PR TITLE
task: moving main.go to cmd/app/main.go

### PR DESCRIPTION
**Pull request by paganotoni**
This is related to #115. It moves the generated application main to the `cmd/app` folder. It ensures that newly generated apps have the main.go file in there and the refresh configuration file references it.

While working on this I wondered if we should also decouple the things we are embedding within the CLI (migrations and tasks) and move these into a separate binary we add in the cmd folder, maybe `cmd/cli`. I've seen some people mention that its Go style to have multiple binaries for the same codebase. But this is part of another conversation.

Changes needed to complete this task:

- [x] Changes on the generation of a new app
- [x] Fixer to move main.go.
- [x] Fixer to update refresh file.
- [ ] Documentation and communication of the change.
  - https://gobuffalo.io/documentation/getting_started/directory-structure/

### CODE REVIEW

**Comment by fasmat** [comment]
I added just a few small comments, that should be easily fixable.

Looks good to me! Thanks for the effort @paganotoni !

**Comment by paganotoni** [comment]
Thanks @fasmat! I don't see those comments yet. I know the Github UI for review sometimes is tricky. Can you confirm you submitted those comments? Appreciate it.

**Comment by fasmat** [comment]
Sorry I hadn't submitted my review. 😅

File: internal/genny/build/bin.go [hunk 1]
```diff
@@ -46,9 +47,14 @@ func buildCmd(opts *Options) (*exec.Cmd, error) {
 	if len(opts.LDFlags) > 0 {
 		flags = append(flags, opts.LDFlags)
 	}
+
 	if len(flags) > 0 {
 		buildArgs = append(buildArgs, "-ldflags", strings.Join(flags, " "))
 	}
 
+	prefix := "." + string(filepath.Separator)
+	path := filepath.Join("cmd", "app")
```
[the remark this answers is not in the document]
Comment (paganotoni) [reply internal/genny/build/bin.go:56]: Yeah. Thats what I was aiming to do but when I do that Go trims the dot from the beginning. See https://go.dev/play/p/9_FtDkLLh6-

**Review by fasmat — APPROVED** [review]

**Review by sio4 — APPROVED** [review]
It looks good to me! I tested a simple generation sequence (buffalo new, dev, build) and it worked well.

**Pull request merged**